app/models/grda_warehouse/postgres_inspector.rb

```diff
@@ -0,0 +1,100 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# == GrdaWarehouse::PostgresInspector
+#
+# Utility to help monitor health and growth of postgres database
+#
+class GrdaWarehouse::PostgresInspector
+  include SafeInspectable
+  delegate :select_all, :quote, to: :connection
+
+  attr_accessor :name, :connection
+
+  def initialize(name:, connection:)
+    self.name = name
+    self.connection = connection
+  end
+
+  def self.inspect_each
+    configs = Rails.configuration.database_configuration[Rails.env]
+
+    configs.each_pair do |name, config|
+      next unless config['adapter'] == 'postgresql'
```

Comment:
not sure it matters, but we have postgis dbs, does that get reflected in the adapter?

Reply:
adjusted for postgis